src/views/dataset/NewDataset.vue

```diff
@@ -75,9 +76,14 @@ function findCommonPrefix(strings: string[]): string {
     return strings[0];
   }
 
+  // Extract the non-metadata prefix of each filename. Note that because of the
+  // way the regex is constructed, the first match group will never be `null`.
+  const re = /(.*?)(_channel|_XY|_Z|_time|$)/;
```

Comment:
With the more complicated regex, this has to be more complicated, too.

Reply:
See d65ef36.